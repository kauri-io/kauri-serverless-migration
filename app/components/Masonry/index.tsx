import React, { Component } from "react";
import styled, { css } from "styled-components";
import theme from "../../lib/theme-config";

const DEFAULT_CARD_WIDTH = theme.DEFAULT_CARD_WIDTH;

const paddingCSS = css`
  padding-top: ${props => props.theme.paddingTop};
  padding-bottom: ${props => props.theme.paddingTop};
  ${props => props.theme.padContent};
`;

const MasonryContainer = styled.div<{ withPadding: boolean }>`
  display: grid;
  width: 100%;
  grid-template-columns: repeat(
    auto-fill,
    minmax(${DEFAULT_CARD_WIDTH}px, ${DEFAULT_CARD_WIDTH}px)
  );
  grid-gap: ${props => props.theme.space[2]}px
    ${props => props.theme.space[3]}px;
  ${props => props.withPadding && paddingCSS};
  @media (max-width: ${props => props.theme.breakpoints[0]}) {
    justify-content: center;
  }
`;

interface IProps {
  withPadding?: boolean;
}

class Masonry extends Component<IProps> {
  public render() {
    const { children, withPadding = true } = this.props;

    return (
      <MasonryContainer withPadding={withPadding}>{children}</MasonryContainer>
    );
  }
}

export default Masonry;
