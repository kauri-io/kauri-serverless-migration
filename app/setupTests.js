import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
