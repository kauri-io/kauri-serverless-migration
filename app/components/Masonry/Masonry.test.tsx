import Masonry from './index'

import { mountWithTheme } from '../../setupTests'

const props = {}

describe('components/Masonry', () => {
    it('snapshot should match', () => {
        const wrapper = mountWithTheme(
            <Masonry {...props}>
                <div>Test</div>
            </Masonry>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
