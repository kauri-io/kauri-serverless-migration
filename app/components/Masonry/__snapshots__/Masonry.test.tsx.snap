// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Masonry should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <Masonry>
    <Masonry__MasonryContainer
      withPadding={true}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Masonry__MasonryContainer-sc-8bsb2v-0",
              "isStatic": false,
              "lastClassName": "eAEBsh",
              "rules": Array [
                "display:grid;width:100%;grid-template-columns:repeat( auto-fill,minmax(",
                "288",
                "px,",
                "288",
                "px) );grid-gap:",
                [Function],
                "px ",
                [Function],
                "px;",
                [Function],
                ";@media (max-width:",
                [Function],
                "){justify-content:center;}",
              ],
            },
            "displayName": "Masonry__MasonryContainer",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Masonry__MasonryContainer-sc-8bsb2v-0",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        withPadding={true}
      >
        <div
          className="Masonry__MasonryContainer-sc-8bsb2v-0 eAEBsh"
        >
          <div>
            Test
          </div>
        </div>
      </StyledComponent>
    </Masonry__MasonryContainer>
  </Masonry>
</ThemeProvider>
`;
